### PR TITLE
Add new quiz flow - unit test

### PR DESCRIPTION
**Pull request by nunibaranes**

### CODE REVIEW

File: src/components/Screen/CourseEditorScreen/QuizSettingsForm/QuizSettingsForm.test.tsx [hunk 1]
```diff
@@ -16,24 +16,70 @@ const state = {
...
+  it('should change quiz.isEnabled default data to false', () => {
     const enableQuizToggle = getSwitchByClosestSelector('.enable-quiz');
     expect(enableQuizToggle).toBeChecked();
-
     if (enableQuizToggle) userEvent.click(enableQuizToggle);
-    await expect(enableQuizToggle).not.toBeChecked();
+    expect(enableQuizToggle).not.toBeChecked();
+  });
+
+  it('should change quiz.forceCourseCompletion default data to true', () => {
```
Comment (dr15) [inline src/components/Screen/CourseEditorScreen/QuizSettingsForm/QuizSettingsForm.test.tsx:42]: Maybe try wording it in a way that explains what the user tried to do

File: src/components/Screen/CourseEditorScreen/__tests__/CourseOutlineQuizFlow.test.tsx [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+  describe('course data with new quiz data', () => {
+    const stateWithQuiz = { ...state, course: { ...state.course, quiz: newQuiz } };
+
+    it('outline should contain new quiz section', () => {
+      renderWithProvider(<CourseOutline />, (stateWithQuiz as unknown) as IState);
+      const quiz = screen.getByText('Quiz').closest('.quiz');
+      expect(quiz).toBeInTheDocument();
+    });
+
+    describe('quiz settings', () => {
```
Comment (dr15) [inline src/components/Screen/CourseEditorScreen/__tests__/CourseOutlineQuizFlow.test.tsx:121]: I would reconsider nesting tests

**Review by dr15 — CHANGES_REQUESTED** [review]
Left a few comments

File: src/components/Screen/CourseEditorScreen/__tests__/CourseOutlineQuizFlow.test.tsx [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    const quiz = screen.getByText('Quiz').closest('.quiz');
+    expect(quiz).toBeInTheDocument();
+  });
+
+  it("click on quiz's header should open quiz settings panel details", async () => {
+    renderWithProvider(<CourseOutline />, (stateWithQuiz as unknown) as IState);
+    await openQuizSettingsForm();
+  });
+
+  it("quiz settings - enable/disable enable quiz switch field should effect quiz's title", async () => {
```
Comment (dr15) [inline src/components/Screen/CourseEditorScreen/__tests__/CourseOutlineQuizFlow.test.tsx:156]: I think there's an extra 'enable' in the title

File: src/components/Screen/CourseEditorScreen/QuizSettingsForm/QuizSettingsForm.test.tsx [hunk 4]
```diff
@@ -16,24 +16,71 @@ const state = {
...
+      expect(passmark).toHaveValue('100');
+
+      userEvent.clear(passmark);
+      userEvent.type(passmark, '80');
+      expect(passmark).toHaveValue('80');
+    }
+  });
+
+  it('click on `random question` switch field should change from off to on', () => {
+    const randQuestions = getSwitchByClosestSelector('.random-questions');
```
Comment (dr15) [inline src/components/Screen/CourseEditorScreen/QuizSettingsForm/QuizSettingsForm.test.tsx:67]: I couldn't find where/how this class is added

**Review by dr15 — CHANGES_REQUESTED** [review]
I  have a few questions

File: src/components/Screen/CourseEditorScreen/__tests__/CourseOutlineQuizFlow.test.tsx (hunk 3 above)
Comment (nunibaranes) [reply src/components/Screen/CourseEditorScreen/__tests__/CourseOutlineQuizFlow.test.tsx:156]: You right :) thanks

File: src/components/Screen/CourseEditorScreen/QuizSettingsForm/QuizSettingsForm.test.tsx (hunk 4 above)
Comment (nunibaranes) [reply src/components/Screen/CourseEditorScreen/QuizSettingsForm/QuizSettingsForm.test.tsx:67]: This class exist in `QuizSettingsForm/index.tsx`. (line 76) :)
I added to each `FormGroup` className.

**Pull request closed without merging**